pkg/server/internal/resultset/resultset.go

```diff
@@ -142,3 +143,25 @@ func (trs *tidbResultSet) FieldTypes() []*types.FieldType {
 func (trs *tidbResultSet) SetPreparedStmt(stmt *core.PlanCacheStmt) {
 	trs.preparedStmt = stmt
 }
+
+// TryDetach creates a new `ResultSet` which doesn't depend on the current session context.
+func (trs *tidbResultSet) TryDetach() (ResultSet, bool, error) {
+	detachableRecordSet, ok := trs.recordSet.(sqlexec.DetachableRecordSet)
+	if !ok {
+		return nil, false, nil
+	}
+
+	recordSet, detached, err := detachableRecordSet.TryDetach()
+	if !detached {
+		return nil, false, nil
+	}
+	if err != nil {
+		return nil, true, err
```

Comment:
why return true here?

Reply:
It follows the second return value from `detachableRecordSet.TryDetach()`. I'll modify it to `return nil, detached, err` to make it clearer. It's always `true` in this case.

Reply:
I realized that the implementation of `execStmtResult.TryDetach()` has a little mistake: it should return `nil, true, err` when facing the error returned by `finishStmt` :facepalm: . I've added a test for this situation.